repour/adjust/util.py

```diff
@@ -236,12 +242,75 @@ async def print_java_version(java_bin_dir=""):
 
 
 async def generate_user_context():
-    """For now, returns a string of key:value,key:value"""
+    # get the current span, created by flask
+    current_span = trace.get_current_span()
+    if current_span.get_span_context().is_valid:
+        trace_id = format_trace_id(current_span.get_span_context().trace_id)
+        span_id = format_span_id(current_span.get_span_context().span_id)
+        trace_flags = current_span.get_span_context().trace_flags
+    else:
+        trace_id = log_util.get_mdc_value_in_task("trace_id")
+        span_id = log_util.get_mdc_value_in_task("span_id")
+        trace_flags = TraceFlags.get_default()
+
+    sampled = (TraceFlags.SAMPLED & trace_flags) != 0
+
+    if trace_id is not None:
+        current_traceparent = "00-{}-{}-{}".format(
+            trace_id, span_id, "01" if sampled else "00"
+        )
+    else:
+        current_traceparent = ""
+
+    """Returns a string of key:value,key:value"""
```

Comment:
Ah that comment should be just below the method definition: it's Python's Javadoc and is placed just below the method rather than above.

Reply:
Moved.